pipeline/inferrer/inference_manager/src/main/scala/uk/ac/wellcome/platform/inference_manager/services/InferrerAdapter.scala

```diff
@@ -72,17 +76,32 @@ object FeatureVectorInferrerAdapter
   def augmentInput(image: MergedImage[Identified],
                    inferrerResponse: Option[InferrerResponse]): AugmentedImage =
     image.augment {
-      inferrerResponse collect {
-        case FeatureVectorInferrerResponse(features, lsh_encoded_features)
-            if features.size == 4096 =>
-          val (features1, features2) = features.splitAt(features.size / 2)
-          InferredData(
-            features1 = features1,
-            features2 = features2,
-            lshEncodedFeatures = lsh_encoded_features
-          )
+      inferrerResponse flatMap {
+        case FeatureVectorInferrerResponse(
+            features_b64,
+            lsh_encoded_features) =>
+          val features = decodeBase64ToFloatList(features_b64)
+          if (features.size == 4096) {
+            val (features1, features2) = features.splitAt(features.size / 2)
+            Some(
+              InferredData(
+                features1 = features1,
+                features2 = features2,
+                lshEncodedFeatures = lsh_encoded_features
+              )
+            )
+          } else None
       }
     }
 
+  private def decodeBase64ToFloatList(base64str: String): List[Float] = {
```

Comment:
I cannot believe how complicated it is to do this in the JVM

Reply:
😢